pkg/machines/components/vcpuModal.jsx

```diff
@@ -0,0 +1,291 @@
+                    />
+                    <InfoRecord
+                        descr={_("vCPU Maximum")}
+                        tooltip={cockpit.format(_("Maximum number of virtual CPUs allocated for the guest OS, which must be between 1 and $0"), this.props.hypervisorMax)}
+                        value={<input id="machines-vcpu-max-field" type="number" className="form-control" onChange={this.onMaxChange} value={this.state.max} />}
+                    />
+                </table>
+            </td>);
+        } else {
+            vcpuColumn = (<td>
+                <table className='form-table-ct'>
+                    <InfoRecord
+                        descr={_("vCPU Count")}
+                        tooltip={_("Number of virtual CPUs that gonna be used.")}
+                        value={<input id="machines-vcpu-count-field" type="number" className="form-control" value={this.state.count} disabled />}
+                    />
+                </table>
+            </td>);
+            sctColumn = (<td>
+                <table className='form-table-ct vcpu-detail-modal-right'>
+                    <InfoRecord descr={_("Sockets")} tooltip={_("Preferred number of sockets to expose to the guest.")} value={
+                        <input id='socketsInput' value={this.state.sockets.toString()} onChange={this.onOvirtChange("sockets")} disabled={this.props.isRunning} />
+                    }/>
+                    <InfoRecord descr={_("Cores per socket")} value={
+                        <input id='coresInput' value={this.state.cores.toString()} onChange={this.onOvirtChange("cores")} disabled={this.props.isRunning}/>
+                    }/>
+                    <InfoRecord descr={_("Threads per cores")} value={
+                        <input id='threadsInput' value={this.state.threads.toString()} onChange={this.onOvirtChange("threads")} disabled={this.props.isRunning}/>
+                    }/>
+                </table>
+            </td>);
+        }
+
+        return (<div className="modal-body">
+            <table className="vcpu-detail-modal-table">
+                <tr>
+                    {vcpuColumn}
+                    {sctColumn}
+                </tr>
+                { caution }
+            </table>
+        </div>);
+    }
+}
+
+VCPUModalBody.propTypes = {
+    vcpus: React.PropTypes.object,
+    isOvirt: React.PropTypes.bool,
+    cpu: React.PropTypes.shape({
+        topology: React.PropTypes.object.isRequired
+    }).isRequired,
+    onChange: React.PropTypes.func.isRequired,
+    hypervisorMax: React.PropTypes.number
+};
+
+export default function ({ vm, dispatch, config }) {
+
+    let state = {};
+    const onStateChange = (st) => {
+        state = Object.assign({}, st);
+    }
+
+    const ovirtConfig = config.providerState && config.providerState.ovirtConfig;
+    const isOvirt = ovirtConfig && !ovirtConfig.loginInProgress;
```

Comment:
The `pkg/machines` shall not access content of `providerState`, there should be no need for it.
The `pkg/machines` should not be aware of `pkg/ovirt` existence, it just respects loose coupling of the Provider implementation.